drivers/flash/soc_flash_nios2_qspi.c

```diff
@@ -79,7 +79,7 @@ static int flash_nios2_qspi_erase(struct device *dev, off_t offset, size_t len)
 	if (((offset + len) > qspi_dev->data_end) ||
 			(0 != (erase_offset &
 			       (NIOS2_WRITE_BLOCK_SIZE - 1)))) {
-		LOG_ERR("erase failed at offset %ld\n", offset);
+		LOG_ERR("erase failed at offset %" PRId32, offset);
```

Comment:
So, this is not a panacea. There's no formatting code for off_t (the type of "offset"). The best suggestion I'd have is to cast it to (long), and keep printing with %ld. That's compatible with both 32 and 64 bit systems. PRId32 might lead to warning even on 32-bit system.